### PR TITLE
Add tool to catch processes execution statistics in Windows

### PR DESCRIPTION
**Pull request by skvl**
This is used to determine most suitable process to inject into.

### CODE REVIEW

File: examples/win-proc-stat.c [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+
+static const char* win_offset_names[__WIN_OFFSETS_MAX][2] = {
+    [KPCR_PRCB] = {"_KPCR", "Prcb" },
+    [KPCR_PRCBDATA] = {"_KPCR", "PrcbData" },
+    [KPRCB_CURRENTTHREAD] = { "_KPRCB", "CurrentThread" },
+    [ETHREAD_CID] = {"_ETHREAD", "Cid" },
+    [CLIENT_ID_UNIQUETHREAD] = {"_CLIENT_ID", "UniqueThread" },
+};
+
+static bool rekall_lookup_in_json(
```
Comment (tklengyel) [inline examples/win-proc-stat.c]: I don't think you should do this manually, there are already functions available in `libvmi_extra.h` that you can use to look up values in the json profiles, which works with either Rekall or Volatility profiles.

File: examples/win-proc-stat.c [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+    g_hash_table_insert(config, g_strdup("os_type"), g_strdup("Windows"));
+    g_hash_table_insert(config, g_strdup("rekall_profile"), g_strdup(rekall_profile));
+
+    /* initialize the libvmi library */
+    vmi_instance_t vmi = NULL;
+    vmi_init_error_t err = VMI_SUCCESS;
+
+    if (VMI_FAILURE ==
+            vmi_init_complete(&vmi, domain, init_flags, NULL,
+                              VMI_CONFIG_GHASHTABLE, config, &err)) {
```
Comment (tklengyel) [inline examples/win-proc-stat.c]: You can pass in the path directly to `vmi_init_complete` now, no need to create the GHashTable here. Just use VMI_CONFIG_JSON_PATH as the identifier. LibVMI can figure out the OS type from the profile too.

Comment (skvl) [reply examples/win-proc-stat.c]: @tklengyel, done!

File: examples/win-proc-stat.c [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+    GHashTable* stats;
+};
+
+/*
+ * Structures and functions to store stat data
+ */
+typedef union pid_tid {
+    struct {
+        uint64_t pid : 32;
+        uint64_t tid : 32;
```
Comment (tklengyel) [inline examples/win-proc-stat.c]: This is bit odd looking, why not just use `uint32_t pid` & `uint32_t tid`?

File: examples/win-proc-stat.c [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+                profile_json,
+                win_offset_names,
+                __WIN_OFFSETS_MAX,
+                offsets))
+        printf("Failed to find offsets for array of structure names and subsymbols.\n");
+
+    return offsets;
+}
+
+static int32_t get_current_thread_id(vmi_instance_t vmi, vmi_event_t* event)
```
Comment (tklengyel) [inline examples/win-proc-stat.c]: I'm afraid you copied a large chunk of this code verbatim from DRAKVUF. You can't do that, the license of DRAKVUF is incompatible with LibVMI's LGPL. I would suggest you close this PR and add this as an extra tool to DRAKVUF using the DRAKVUF license.

**Pull request closed without merging**